src/SkeletalTrapezoidation.cpp

```diff
@@ -0,0 +1,3083 @@
+        // starting_edge->prev = nullptr;
+//         starting_edge->from->data.distance_to_boundary = 0; // TODO
+
+        make_rib(prev_edge, start_source_point, end_source_point, true);
+        for (vd_t::edge_type* vd_edge = starting_vd_edge->next(); vd_edge != ending_vd_edge; vd_edge = vd_edge->next())
+        {
+            assert(vd_edge->is_finite());
+            Point v1 = VoronoiUtils::p(vd_edge->vertex0());
+            Point v2 = VoronoiUtils::p(vd_edge->vertex1());
+            transfer_edge(v1, v2, *vd_edge, prev_edge, start_source_point, end_source_point, points, segments);
+
+            make_rib(prev_edge, start_source_point, end_source_point, vd_edge->next() == ending_vd_edge);
+        }
+
+        transfer_edge(VoronoiUtils::p(ending_vd_edge->vertex0()), end_source_point, *ending_vd_edge, prev_edge, start_source_point, end_source_point, points, segments);
+        // ending_edge->next = nullptr;
+        prev_edge->to->data.distance_to_boundary = 0;
+
+        debugCheckGraphConsistency(true);
+    }
+
+    
+    debugCheckGraphCompleteness();
+    debugCheckGraphConsistency();
+    debugCheckGraphExistance();
+
+    separatePointyQuadEndNodes();
+
+    debugCheckGraphCompleteness();
+    debugCheckGraphConsistency();
+    debugCheckGraphExistance();
+
+    fixNodeDuplication();
+
+    debugCheckGraphCompleteness();
+    debugCheckGraphConsistency();
+    debugCheckGraphExistance();
+    debugCheckEndpointUniqueness();
+    
+    collapseSmallEdges();
+
+
+    debugCheckGraphCompleteness();
+    debugCheckGraphConsistency();
+    debugCheckGraphExistance();
+
+    { // set [some_edge] the the first possible edge
+        // that way we can iterate over all reachable edges from node.some_edge without needing to iterate backward
+        for (edge_t& edge : graph.edges)
+        {
+            if (!edge.prev)
+            {
+                edge.from->some_edge = &edge;
+            }
+        }
+    }
+
+
+    debugCheckGraphCompleteness();
+    debugCheckGraphConsistency();
+    debugCheckGraphStructure();
+    debugCheckGraphReachability();
+    debugCheckGraphExistance();
+    
+#ifdef DEBUG
+    {
+        AABB aabb(polys);
+        SVG svg("output/vq.svg", aabb);
+        debugOutput(svg, false, false);
+    }
+    {
+        AABB aabb(polys);
+        SVG svg("output/radial_dists.svg", aabb);
+        debugOutput(svg, false, true);
+    }
+    {
+        AABB aabb(polys);
+        SVG svg("output/bead_counts.svg", aabb);
+        debugOutput(svg, false, false, true);
+    }
+    {
+        AABB aabb(polys);
+        SVG svg("output/locations.svg", aabb);
+        debugOutput(svg, false, false, false, true);
+    }
+    debugCheckGraphCompleteness();
+    debugCheckGraphConsistency();
+
+    if (generate_MAT_STL)
+    {
+        STLwriter stl("output/mat.stl");
+        debugOutput(stl);
+    }
+#endif
+
+    vd_edge_to_he_edge.clear();
+    vd_node_to_he_node.clear();
```

Comment:
Why clear it? It's not going to release any memory, you know.

`clear` is not guaranteed to release any memory and for GCC it releases none. It's going to stay in the data structure's allocator. If you know that the container is not going to be used any more, you can `std::swap` it with an empty data structure. This will release the memory to the program's allocator (but will still not reduce memory usage).

Reply:
Someone could call `constructFromPolygons` more than once? Placing those clears at the top of the method will make that a bit more clear though.